### PR TITLE
`crux-mir`: `symbolic_f{16,32,64,128}` overrides, `slice::fill` test case

### PR DESCRIPTION
**Pull request by RyanGlScott**
This bumps the `mir-json` submodule to bring in the changes from https://github.com/GaloisInc/mir-json/pull/303 and https://github.com/GaloisInc/mir-json/pull/300 and adds corresponding test cases. The former change requires adding `crux-mir` overrides to support the newly added `symbolic_f{16,32,64,128}` functions.

### CODE REVIEW

File: crux-mir/test/symb_eval/float/symbolic.rs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#![feature(f16)]
+#![feature(f128)]
+
+extern crate crucible;
+use crucible::*;
+
+fn symbolic_test<S: PartialEq + Symbolic>() -> S {
```
Comment (podhrmic) [inline crux-mir/test/symb_eval/float/symbolic.rs:7]: Would it be appropriate to have another, more interesting test that actually requires some computation with the float?

E.g. `crucible_assert( (x+x) == 2x)` or something like that, properly handling the edge cases (zero, NaN, maybe something else)? 

Or should such test be in a different directory?

**Review by podhrmic — COMMENTED** [review]
Thanks for getting this done so quickly! My only comment is about the depth of the float test (see above).

File: crux-mir/test/symb_eval/float/symbolic.rs (hunk 1 above)
Comment (RyanGlScott) [reply crux-mir/test/symb_eval/float/symbolic.rs:7]: My inclination would be to wait until https://github.com/GaloisInc/crucible/issues/1865 is addressed before doing this. Currently, `crucible-mir`'s representation of floats is somewhat unsound, and a theorem like `x+x == 2*x` actually _shouldn't_ hold if `x` is NaN, but `crucible-mir` won't detect this at the moment. We'd need to switch `crucible-mir`'s representation from `RealValType` (i.e., algebraic reals) to `FloatType` (i.e., IEEE floats) in order for `crucible-mir` to be able to do this properly.

Comment (samcowger) [reply crux-mir/test/symb_eval/float/symbolic.rs:7]: Given that `crucible-mir`'s floating-point machinery is already unsound, I wonder whether we should avoid adding more interfaces for it before we fix it, perhaps via #1865. What do you think?

Comment (RyanGlScott) [reply crux-mir/test/symb_eval/float/symbolic.rs:7]: Coincidentally, I am working on a fix for #1865 as we speak. I'm happy to submit another PR (based on this one) and merge everything en masse once we're happy with it.

Comment (RyanGlScott) [reply crux-mir/test/symb_eval/float/symbolic.rs:7]: See https://github.com/GaloisInc/crucible/pull/1871.
